docs/design/resources.md

```diff
@@ -0,0 +1,96 @@
+# Operator Resources
+
+The etcd cluster operator extends the Kubernetes API with two new
+resource definitions:
+
+* `EtcdCluster`, or the "cluster resource", defines an entire
+  operating cluster.
+* `EtcdPeer`, or the "peer resource", defines a single peer of the
+  etcd cluster. This principally concerns a Pod, but may also involve
+  attached storage and network identity.
+
+## The Peer Resource
+
+The peer resource represents a single peer in an etcd
+cluster. Encoding it's bootstrap instructions, persistence settings,
+and extra pod settings (e.g., overridden CPU limits). Its `status`
+field exposes information about the peer including its hostname,
+liveness status, the peer's current generation, and if it's currently
+the leader of the etcd cluster.
```

Comment:
+1 to this, it's definitely where I see the EtcdPeer API going. That said, I'm not certain we'll implement all of this from day zero. Perhaps prefixing this sentence with "and *will* be used to expose various information about the peer, including but not limited to ..."?

Reply:
I'm also not sure if generation is something we're going to use... I'm not sure why we do need it, nor what we'd define it as. Perhaps easiest to omit for now?

Reply:
I was actually meaning the raft concept of a leader "term", but I don't know if etcd actually exposes that. I'll remove that example and clarify to "and may in future".

Reply:
I think it does, and this may be useful in future, but I'd rather us err on the side of less features initially